### PR TITLE
Make band scale default for bar

### LINKED ISSUE #1615: band scale doesn't support size yet
With Vega 3, we can set `band` to 0.5 to center the bandSize 

See `bandPosition()` in `mixins.ts`.

### LINKED ISSUE #2161: Default scale for bar
Consider if we want to use band scale by default.

### PR DESCRIPTION
**Pull request by kanitw**
Fix #1615 and fix #2161

### CODE REVIEW

**Comment by domoritz** [comment]
What's missing in this PR?

**Comment by kanitw** [comment]
@domoritz can't remember. I'll take care of it later.

File: src/compile/mark/mixins.ts [hunk 1]
```diff
@@ -91,10 +92,36 @@ export function text(model: UnitModel, vgChannel: 'text' | 'tooltip' = 'text') {
...
+    if (orient) {
+      const centeredBandPositionMixins = {
+        // Use xc/yc and place the mark at the middle of the band
+        // This way we never have to deal with size's condition for x/y position.
+        [channel+'c']: ref.fieldRef(fieldDef, scaleName, {}, ref.band(scaleName, 0.5))
+      };
+
+      if (isFieldDef(model.encoding.size)) {
+        log.warn(log.message.cannotUseSizeFieldWithBandSize(channel));
+        // TODO: apply size to band and set scale range to some values between 0-1.
```
Comment (kanitw) [inline src/compile/mark/mixins.ts]: I was looking into this but it requires some amount of refactoring in scale.  
Given this is a bad encoding anyway, I will focus on more important issues.

Comment (domoritz) [inline src/compile/mark/mixins.ts]: Sounds good. What about https://github.com/vega/vega/issues/845#issuecomment-306538369?

Comment (kanitw) [inline src/compile/mark/mixins.ts]: Done https://github.com/vega/vega-lite/pull/2309/commits/00d74682bfeac9a28edd8049a42e883a676ba9e9

**Pull request merged**